src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BeeWingListener.java

```diff
@@ -0,0 +1,54 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
+import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+/**
+ * This {@link Listener} is responsible for the slow falling effect given to the player
+ * When Nearing the ground.
+ *
+ * @author beSnow
+ */
+public class BeeWingListener implements Listener {
+
+    public BeeWingListener(SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onApproachGround(PlayerMoveEvent e) {
```

Comment:
Using the PlayerMoveEvent is very risky as it can cause a ton of lag.

Reply:
Yeah, i should have been more concernded about the performance.
I'm currently looking for an alternative to PlayerMoveEvent.

Reply:
you can give the potion effect slow falling?